### PR TITLE
chore: adds API for auth integ tests + minor fixes

### PR DESCRIPTION
**Pull request by haverchuck**
*Description of changes:*
Adds adminCreateUser to auth integ test and a few minor test fixes.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by fjnoyp — APPROVED** [review]

**Comment by ragingsquirrel3** [comment]
Question: As auth integ tests now require API, would it be possible to update the resource provisioning to reflect how to create that part of backend? Could either update existing provision script at https://github.com/aws-amplify/amplify-flutter/blob/main/packages/amplify_auth_cognito/example/tool/provision_integration_test_resources.sh#L24, see api example https://github.com/aws-amplify/amplify-flutter/blob/main/packages/amplify_api/example/tool/provision_integration_test_resources.sh#L24, or checkin the the partial amplify dir to this test so it can use pull. Or, could update README to explain how to provision and I can update the provisioning some other time as I've been meaning to checkin the amplify dir for this test.

File: packages/amplify_auth_cognito/example/tool/provision_integration_test_resources.sh [hunk 1]
```diff
@@ -17,9 +17,58 @@ FRONTEND="{\
...
+    echo "We will attempt to create the role."
+}
+
+# put lambda code into bucket
+echo "adding lambda code to S3..."
+aws s3api put-object --bucket $deploymentBucket --key $s3Key --body ./tool/adminCreateUserLambda/adminCreateUser.zip
+
+
+# create lambda function for adminCreateUser
+echo "creating lanmbda with cloudformation..."
```
Comment (ragingsquirrel3) [inline packages/amplify_auth_cognito/example/tool/provision_integration_test_resources.sh:69]: nit: typo

File: packages/amplify_auth_cognito/example/tool/provision_integration_test_resources.sh [hunk 2]
```diff
@@ -17,9 +17,65 @@ FRONTEND="{\
 \"config\":$FLUTTERCONFIG\
 }"
 
+# read the request template and the schema
```
Comment (ragingsquirrel3) [inline packages/amplify_auth_cognito/example/tool/provision_integration_test_resources.sh:20]: I was noticing that this script works well when I run on normal aws account w default profile, but it does not work well with an isengard account or other profile. To account for that, I have a request: make this script accept an env variable AWS_PROFILE, then have the Amplify headless and AWS CLI refer to that variable (default to "default").

At the top, will look like 
```
# In development, AWS_PROFILE should be set. In CI, it's not.
[ "$AWS_PROFILE" ] && profileName="$AWS_PROFILE" || profileName="default";

FLUTTERCONFIG="{\
\"ResDir\":\"./lib/\",\
}"

AMPLIFY="{\
\"projectName\":\"amplifyauthinteg\",\
\"envName\":\"test\",\
\"defaultEditor\":\"code\"\
}"

FRONTEND="{\
\"frontend\":\"flutter\",\
\"config\":$FLUTTERCONFIG\
}"

AWSCLOUDFORMATIONCONFIG="{\
\"configLevel\":\"project\",\
\"useProfile\":"true",\
\"profileName\":\"$profileName\",\
\"region\":\"us-west-2\"\
}"
PROVIDERS="{\
\"awscloudformation\":$AWSCLOUDFORMATIONCONFIG\
}"
```

Then should use that profile as a `--profile` argument to all the AWS CLI commands to ensure everything goes through same profile.

File: packages/amplify_auth_cognito/example/tool/provision_integration_test_resources.sh [hunk 3]
```diff
@@ -17,9 +17,65 @@ FRONTEND="{\
...
+# escape quotes and remove new lines from schema
+schema=${schema//$'"'/'\"'}
+schema=${schema//$'\n'/}
+
+# create the request with the actual schema
+request="${requestTemplate/<SCHEMA_PLACEHOLDER>/$schema}"
+
 amplify init \
 --amplify $AMPLIFY \
 --frontend $FRONTEND \
```
Comment (ragingsquirrel3) [inline packages/amplify_auth_cognito/example/tool/provision_integration_test_resources.sh:33]: will need to add here:
```
--providers $PROVIDERS \
```

File: packages/amplify_auth_cognito/example/tool/provision_integration_test_resources.sh [hunk 4]
```diff
@@ -17,9 +17,65 @@ FRONTEND="{\
...
+# put lambda code into bucket
+echo "adding lambda code to S3..."
+aws s3api put-object --bucket $deploymentBucket --key $s3Key --body ./tool/adminCreateUserLambda/adminCreateUser.zip
+
+# remove zip file
+rm ./tool/adminCreateUserLambda/adminCreateUser.zip
+
+# create lambda function for adminCreateUser
+echo "creating lambda with cloudformation..."
+aws cloudformation deploy --template-file ./tool/adminCreateUserLambda/cloudformation.json --stack-name admin-create-user-stack --parameter-overrides deploymentBucketName=$deploymentBucket s3Key=$s3Key env=test authauthintegrationtestUserPoolId=$userpoolId stackId=$stackId --capabilities CAPABILITY_NAMED_IAM --profile default
```
Comment (ragingsquirrel3) [inline packages/amplify_auth_cognito/example/tool/provision_integration_test_resources.sh:77]: cannot use default profile here, need to pass the `$profileName` variable from above, same for all aws commands in this file.

**Review by ragingsquirrel3 — CHANGES_REQUESTED** [review]
This is looking pretty good! I left some comments about how to improve the provision script to run with different AWS profiles as I think this will be needed given amplify headless commands and aws CLI commands can resolve to different accounts in some circumstances. 

I can discuss it more offline as I debugged it locally a bit.

**Comment by ragingsquirrel3** [comment]
nit request: Would it be possible to update the provision instructions in contributing.md as I think there are some instructions for manually editing the auth backend that are no longer necessary with this change?

File: packages/amplify_auth_cognito/example/tool/provision_integration_test_resources.sh [hunk 5]
```diff
@@ -2,6 +2,9 @@
 set -e
 IFS='|'
 
+# In development, AWS_PROFILE should be set. In CI, it's not.
```
Comment (ragingsquirrel3) [inline packages/amplify_auth_cognito/example/tool/provision_integration_test_resources.sh:5]: nit: please remove last bit about CI or just remove this comment as we will not run this in CI. (my bad as I suggested this initially)

File: packages/amplify_auth_cognito/example/tool/provision_integration_test_resources.sh [hunk 6]
```diff
@@ -17,9 +20,76 @@ FRONTEND="{\
...
+# put lambda code into bucket
+echo "adding lambda code to S3..."
+aws s3api put-object --profile=$profileName --bucket $deploymentBucket --key $s3Key --body ./tool/adminCreateUserLambda/adminCreateUser.zip
+
+# remove zip file
+rm ./tool/adminCreateUserLambda/adminCreateUser.zip
+
+# create lambda function for adminCreateUser
+echo "creating lambda with cloudformation..."
+aws cloudformation deploy --profile=$profileName --template-file ./tool/adminCreateUserLambda/cloudformation.json --stack-name admin-create-user-stack --parameter-overrides deploymentBucketName=$deploymentBucket s3Key=$s3Key env=test authauthintegrationtestUserPoolId=$userpoolId stackId=$stackId --capabilities CAPABILITY_NAMED_IAM --profile default
```
Comment (ragingsquirrel3) [inline packages/amplify_auth_cognito/example/tool/provision_integration_test_resources.sh:91]: This line still refers to default profile at the end.

**Review by ragingsquirrel3 — APPROVED** [review]

File: packages/amplify_auth_cognito/example/tool/provision_integration_test_resources.sh (hunk 6 above)
Comment (haverchuck) [reply packages/amplify_auth_cognito/example/tool/provision_integration_test_resources.sh:91]: lol, so it does. fixing.

**Pull request merged**